source/custom/nimxlcterminaladaptor_restricted_converters.h

```diff
@@ -16,13 +16,11 @@ namespace nimxlcterminaladaptor_restricted_grpc {
 // Add underscore to usings so they don't conflict with including files in the same namespace.
 namespace pb_ = ::google::protobuf;
 
-bool nierr_noOpReallocJson(struct nierr_Status * s, uint32_t size);
-
-void nierr_Status_initialize(struct nierr_Status * status);
-
+template <typename TServerContext>
 struct NIErrStatusOutputConverter {
-  NIErrStatusOutputConverter()
+  NIErrStatusOutputConverter(TServerContext *serverContext)
   {
+    context = serverContext;
     nierr_Status_initialize(&status);
```

Comment:
`nierr_Status` is a C struct that owns a string buffer. Code that creates `nierr_Status` directly is responsible for freeing this buffer. You should either:
- Add a destructor to `NIErrStatusOutputConverter` that calls `nierr_Status_jsonFree`.
- Fork a subset of the C++ wrapper class `nierr::Status` and use that in place of `nierr_Status`.